src/raven.js

```diff
@@ -1988,35 +1990,59 @@ Raven.prototype = {
     }
 
     var url = this._globalEndpoint;
-    (globalOptions.transport || this._makeRequest).call(this, {
-      url: url,
-      auth: auth,
-      data: data,
-      options: globalOptions,
-      onSuccess: function success() {
-        self._resetBackoff();
-
-        self._triggerEvent('success', {
-          data: data,
-          src: url
-        });
-        callback && callback();
-      },
-      onError: function failure(error) {
-        self._logDebug('error', 'Raven transport failed to send: ', error);
 
-        if (error.request) {
-          self._setBackoffState(error.request);
-        }
+    try {
+      (globalOptions.transport || this._makeRequest).call(this, {
+        url: url,
+        auth: auth,
+        data: data,
+        options: globalOptions,
+        onSuccess: function success() {
+          self._resetBackoff();
+
+          self._triggerEvent('success', {
+            data: data,
+            src: url
+          });
+          callback && callback();
+        },
+        onError: function failure(error) {
+          self._logDebug('error', 'Raven transport failed to send: ', error);
 
-        self._triggerEvent('failure', {
-          data: data,
-          src: url
-        });
-        error = error || new Error('Raven send failed (no additional details provided)');
-        callback && callback(error);
-      }
-    });
+          if (error.request) {
+            self._setBackoffState(error.request);
+          }
+
+          self._triggerEvent('failure', {
+            data: data,
+            src: url
+          });
+          error =
+            error || new Error('Raven send failed (no additional details provided)');
+          callback && callback(error);
+        }
+      });
+    } catch (e) {
+      // Something went wrong while invoking either default or user-provided transport function
+      // that should send previously caught exception
+      //
+      // We catch this error, try to send it to Sentry so that user knows what went wrong
+      // and then rethrow original exception so it's not magically swallowed,
+      // while making sure it's not caught again by our code
+
+      // `setTimeout` is required in order for `captureException` below not accidentally trigger
+      // this exact block again and fall into infinite loop.
+      // `e !== lastException` check will only work, once first exception
+      // is actually assigned to our private variable
+      self._originalSetTimeout(function() {
+        var lastException = self.lastException();
+        if (e !== lastException) {
```

Comment:
lastException is only set in captureException, but _sendProcessedPayload is used from captureMessage as well. I'm guessing this is ok because an error from inside raven wouldn't be part of an infinite loop?

Reply:
for instance, if captureMessage resulted in an error thrown from the transport, this could rethrow a previous exception, right? that seems bad

Reply:
also, could this check fail if two errors are sent through in quick succession? don't have an immediate solution for that scenario but I could imagine an infinite loop as a result if it goes ABABABABABAB